tests/functional/upgrade/test_non_stretch_cluster_upgrade.py

```
@@ -0,0 +1,529 @@
+    log.info(f"md5sums captured for {[p.name for p in pod_obj_list]}: {md5sums}")
+    return pod_obj_list, md5sums
+
+
+def _wait_for_migration_off_node(pod_obj_list, outage_node_name, label="workloads"):
+    """
+    Poll until every pod in *pod_obj_list* has rescheduled off *outage_node_name*.
+
+    Returns the refreshed list of migrated pods, or raises TimeoutExpiredError.
+    """
+
+    def _all_rescheduled():
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_all_rescheduled`

Add return type annotation: `bool`

(ANN202)